bzt/modules/reporting.py

```diff
@@ -265,88 +265,121 @@ class JUnitXMLReporter(Reporter, AggregatorListener):
     A reporter that exports results in Jenkins JUnit XML format.
     """
 
-    REPORT_FILE_NAME = "xunit"
-    REPORT_FILE_EXT = ".xml"
-
     def __init__(self):
         super(JUnitXMLReporter, self).__init__()
-        self.report_file_path = "xunit.xml"
         self.last_second = None
+        self.report_file_path = None
 
     def prepare(self):
         """
-        create artifacts, parse options.
-        report filename from parameters
-        :return:
+        create artifacts, parse options, take report filename from parameters
         """
-
-        filename = self.parameters.get("filename", None)
-        if filename:
-            self.report_file_path = filename
-        else:
-            self.report_file_path = self.engine.create_artifact(JUnitXMLReporter.REPORT_FILE_NAME,
-                                                                JUnitXMLReporter.REPORT_FILE_EXT)
-        self.parameters["filename"] = self.report_file_path
-
         if isinstance(self.engine.aggregator, ResultsProvider):
             self.engine.aggregator.add_listener(self)
 
     def aggregated_second(self, data):
-        """
-        :param data:
-        :return:
-        """
         self.last_second = data
 
     def post_process(self):
         """
         Get report data, generate xml report.
         """
-        super(JUnitXMLReporter, self).post_process()
+        filename = self.parameters.get("filename", None)
+        if not filename:
+            filename = self.engine.create_artifact(XUnitFileWriter.REPORT_FILE_NAME, XUnitFileWriter.REPORT_FILE_EXT)
+        self.parameters["filename"] = filename  # reflect it in effective config
+
         test_data_source = self.parameters.get("data-source", "sample-labels")
 
         if test_data_source == "sample-labels":
-            if self.last_second:
-                root_element = self.process_sample_labels()
-                self.save_report(root_element)
-            else:
+            if not self.last_second:
                 self.log.warning("No last second data to generate XUnit.xml")
+            else:
+                writer = XUnitFileWriter(self.engine, 'sample_labels')
+                self.process_sample_labels(writer)
+                writer.save_report(filename)
         elif test_data_source == "pass-fail":
-            root_element = self.process_pass_fail()
-            self.save_report(root_element)
+            self.log.warning("Using 'data-source=pass-fail' is deprecated and will be removed in future release. "
+                             "Use pass-fail module's own option")
```

Comment:
I just forgot to remove the message. For now, we will keep old approach with separate junit-xml module.